include/cop0.h

```diff
@@ -14,7 +14,7 @@
 
 /** @brief Read the COP0 Count register (see also TICKS_READ). */
 #define C0_COUNT() ({ \
-    uint32_t x; \
+    volatile uint32_t x; \
```

Comment:
Uhm I have many doubt about this. I've kept this in my tree for a long time, but I'm always unable to really reproduce the bug that is fixed by this change. Also, the change the way it is written pessimizes the code a little bit. So maybe I should back this out from this PR, and try first to reproduce the bug.

Reply:
I used explicit dependencies for the tick test's inlines but it only works if the referenced symbol is used on the code path that we want to execute after/before it. GCC docs clearly states that it can change inline assembly's position depending on its side effects and the only order guarantee is for the instructions within the same inline block. I'm not sure clobbers will fix this either b/c it might still be placed differently if some instructions do not work on data though I'm not sure if there are such practical examples. I agree that we need to find a way to reproduce and then commit to a solution.

The timing issue between ed/64drive is eventually solved so I think the same will happen to this one as well.

Reply:
Hmm without explicit dependencies it will always be possible to switch instructions wth syscalls/breaks for example